rewriter/Minitest.cc

```diff
@@ -420,28 +542,99 @@ ast::ExpressionPtr runSingle(core::MutableContext ctx, bool isClass, ast::Send *
         return constantMover.addConstantsToExpression(send->loc, move(method));
     }
 
+    if (send->numPosArgs() == 0 && send->fun == core::Names::subject()) {
+        auto declLoc = send->funLoc;
+        return ast::MK::SyntheticMethod0(send->loc, declLoc, core::Names::subject(), std::move(block->body));
+    }
+
+    // Handle include_context with block (1+ pos args + optional kwargs/block)
+    if (send->fun == core::Names::includeContext() && send->numPosArgs() >= 1) {
+        auto &arg = send->getPosArg(0);
+        auto argString = to_s(ctx, arg);
+        // Create an include statement that includes the filtered shared_examples module
+        // Use proper scoping - if we're inside a describe block, look locally; otherwise use root
+        auto moduleName = ast::MK::UnresolvedConstant(
+            arg.loc(), insideDescribe ? ast::MK::EmptyTree() : ast::MK::Constant(arg.loc(), core::Symbols::root()),
+            ctx.state.enterNameConstant("<shared_examples_module '" + argString + "'>"));
```

Comment:
ditto my earlier comment

https://github.com/sorbet/sorbet/pull/9119/files#r2283780722

also, it looks like you have this code duplicated 5 times. It should probably be in a helper function.

Reply:
updated fdb81cebb